handlers/yo/handler.go

```diff
@@ -130,40 +118,31 @@ func (h *handler) SendLegacy(ctx context.Context, msg courier.MsgOut, clog *cour
 
 			req, err := http.NewRequest(http.MethodGet, sendURL.String(), nil)
 			if err != nil {
-				return nil, err
+				return courier.ErrConnectionFailed
```

Comment:
these errs.. that probably can't happen can be just returned as `err` and they'll be treated as an internal error